chapters/en/slides/module5/module5_06.Rmd

```diff
@@ -0,0 +1,288 @@
+---
+params:
+  dynamictitle: "module5_06"
+title: "`r params$dynamictitle`"
+output:
+  md_document:
+    variant: gfm
+    pandoc_args: "--markdown-headings=atx"
+---
+
+```{r setup, include=FALSE}
+source('../../../../slide-setup.R')
+# Remember to also manually update the YAML title above
+```
+
+type: slides
+
+
+# Annotating with text and color
+
+Notes:
+In this module we will be learning how to use color and text
+to annotate our visualizations
+and highlight important features.
+
+---
+
+## Understanding the periodic variations in the Wikipedia donations
+
+```{python}
+# TODO save wrangled data?
+import pandas as pd
+import altair as alt
+from vega_datasets import data
+
+donations = (
+    pd.read_csv("https://frdata.wikimedia.org/donationdata-vs-day.csv", parse_dates=['date'])
+    .query('2020 < date < 2021')
+    .assign(week_day=lambda df: df['date'].dt.day_name().str[:3]))
+
+alt.Chart(donations, title='Wikipedia donations peak during Christmas').mark_line().encode(
+    alt.X('date', title=None),
+    alt.Y('sum', axis=alt.Axis(format='$s'), title='Amount donated per day'))
+```
+
+Notes:
+Recall this plot of Wikipedia donations from a previous slide deck in this module.
+We noted that the oscillating pattern might indicate 
+a dependency on a periodic date feature,
+such as weekday or day of the month.
+
+---
+
+## Most contributions are donated in the middle of the week
+
+```{python}
+from calendar import day_abbr
+
+
+alt.Chart(donations).mark_bar().encode(
+    alt.Y('week_day', sort=list(day_abbr), title=None),
+    alt.X('sum(sum)', axis=alt.Axis(format='$s'), title='Total donated amount'))
+```
+
+Notes:
+Here we are using the built-in calendar module
+to get a sorted list of the weekday abbreviations,
+but we could also have written it out by hand.
+
+We can see that most contributions seem to be donated in the middle of the week
+and the Wed stands out as the day
+where the highest amount of money is donated.
+
+To draw additional attention to this bar,
+we could use color to highlight it.
+
+---
+
+## Color can be used to highlight specific features of a visualization
+
+```{python}
+alt.Chart(donations).mark_bar().encode(
+    alt.Y('week_day', sort=list(day_abbr), title=None),
+    alt.X('sum(sum)', axis=alt.Axis(format='$s'), title='Total donated amount'),
+    color = alt.condition(alt.datum.week_day == 'Wed',
+                          alt.value('coral'),
+                          alt.value('steelblue')))
```

Comment:
Can we not have a condition that looks for the highest bar? What if we wanted to reuse this code in the future and Wed was not the highest/longest bar?

Reply:
Good point! There is none am aware of in vegalite but I  can use pandas